### PR TITLE
Add Orders Get Single endpoint with integration tests

### PR DESCRIPTION
**Pull request by badekk**
## Add Orders Get Single endpoint with integration tests

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## How has this been tested?

- [ ] Unit test(s)
- [x] Integration test(s)
- [ ] Tested manually

## Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] My changes generate no new warnings
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes
- [X] I have checked my code and corrected any misspellings
- [ ] I have added frontend code and checked it for mobile & desktop

### CODE REVIEW

**Comment by mar-pan** [comment]
Task linked: [CU-862jekz07 Get single](https://app.clickup.com/t/862jekz07)

File: src/backend/Order/Actions/OrdersActions/OrdersGetSingle/OrdersGetSingleService.cs [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+                    .ThenInclude(c => c.Product)
+                .SingleOrDefaultAsync(c => c.OwnerUserId == currentUser.Id && c.Id == decodedOrderId);
+
+            if (dbOrderResult == null)
+            {
+                _logger.LogError("Couldn't find Order with id:{decodedOrderId} in database", decodedOrderId);
+                throw new ItemNotFoundException("Sorry we coudln't find that order in database");
+            }
+
+            var dbDeliveryResult = await _appDbContext.Deliveries
```
Comment (h4570) [inline src/backend/Order/Actions/OrdersActions/OrdersGetSingle/OrdersGetSingleService.cs:58]: A czemu nie Include w Orders?

File: src/backend/Order/Actions/OrdersActions/OrdersGetSingle/OrdersGetSingleService.cs [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+                Id = _hashids.EncodeLong(dbOrderResult.Id),
+                Name = dbOrderResult.Name,
+                Percentage = dbOrderResult.Percentage,
+                CreationDate = dbOrderResult.CreationDate,
+                CountryName = dbOrderResult.Address?.Country?.EnglishName!,
+                GpsLatitude = dbOrderResult.Address!.GpsLatitude,
+                GpsLongitude = dbOrderResult.Address.GpsLongitude,
+                City = dbOrderResult.Address?.City!,
+                PostalCode = dbOrderResult.Address?.PostalCode!,
+                Street = dbOrderResult.Address?.Street!,
```
Comment (h4570) [inline src/backend/Order/Actions/OrdersActions/OrdersGetSingle/OrdersGetSingleService.cs:93]: Zmień nazwę property na FullStreet i użyj buildera. Tak samo jak w create-order/addresses

**Review by h4570 — CHANGES_REQUESTED** [review]

File: src/backend/Order/Actions/OrdersActions/OrdersGetSingle/OrdersGetSingleService.cs (hunk 1 above)
Comment (badekk) [reply src/backend/Order/Actions/OrdersActions/OrdersGetSingle/OrdersGetSingleService.cs:58]: hmm.. chyba Listy zapomniałem dodać do Orders ;x bo nie mam jak z Orders do Delivery się dostać.

File: src/backend/Order/Actions/OrdersActions/OrdersGetSingle/OrdersGetSingleService.cs [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+                .Include(c => c.Address)
+                    .ThenInclude(c => c!.Country)
+                .Include(c => c.OrderProducts)!
+                    .ThenInclude(c => c.Product)
+                .SingleOrDefaultAsync(c => c.OwnerUserId == currentUser.Id && c.Id == decodedOrderId);
+
+            if (dbOrderResult == null)
+            {
+                _logger.LogError("Couldn't find Order with id:{decodedOrderId} in database", decodedOrderId);
+                throw new ItemNotFoundException("Sorry we coudln't find that order in database");
```
[the remark this answers is not in the document]
Comment (badekk) [reply src/backend/Order/Actions/OrdersActions/OrdersGetSingle/OrdersGetSingleService.cs:55]: fixed

File: src/backend/Order/Actions/OrdersActions/OrdersGetSingle/OrdersGetSingleService.cs [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+            }
+
+            var dbDeliveryResult = await _appDbContext.Deliveries
+                .Where(c => c.OrderId == decodedOrderId)
+                .ToListAsync();
+
+            if (!dbDeliveryResult.Any())
```
[the remark this answers is not in the document]
Comment (badekk) [reply src/backend/Order/Actions/OrdersActions/OrdersGetSingle/OrdersGetSingleService.cs:62]: mb, fixed

File: src/backend/Order/Actions/OrdersActions/OrdersGetSingle/OrdersGetSingleService.cs (hunk 2 above)
Comment (badekk) [reply src/backend/Order/Actions/OrdersActions/OrdersGetSingle/OrdersGetSingleService.cs]: fixed

**Pull request merged**